quic/client/handshake/ClientHandshakeFactory.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace quic {
+
+class ClientHandshake;
+struct QuicCryptoState;
+
+class ClientHandshakeFactory {
+ public:
+  virtual std::unique_ptr<ClientHandshake> makeClientHandshake(
+      QuicCryptoState& cryptoState) = 0;
+
+  virtual ~ClientHandshakeFactory() = default;
```

Comment:
conventionally in the codebase, the virtual dtor is the first thing in the class